src/neo/Helper.cs

```diff
@@ -310,5 +310,88 @@ internal static BigInteger WeightedAverage<T>(this IEnumerable<T> source, Func<T
                 yield return resultSelector(item, weight);
             }
         }
+
+        public static byte[] Concat(this byte[] a, byte[] b)
+        {
+            var result = new byte[a.Length + b.Length];
+            a.CopyTo(result, 0);
+            b.CopyTo(result, a.Length);
+            return result;
+        }
+
+        public static byte[] CommonPrefix(this byte[] a, byte[] b)
+        {
+            var prefix = Array.Empty<byte>();
+            var minLen = a.Length <= b.Length ? a.Length : b.Length;
+
+            if (a.Length != 0 && b.Length != 0)
+            {
+                for (int i = 0; i < minLen; i++)
+                {
+                    if (a[i] != b[i]) break;
+                    prefix = prefix.Add(a[i]);
+                }
+            }
+            return prefix;
+        }
+
+        public static bool Equal(this byte[] a, byte[] b)
+        {
+            if (a is null || b is null) return false;
+            var len = a.Length;
+            if (len != b.Length) return false;
+            unsafe
+            {
+                fixed (byte* ap = a, bp = b)
+                {
+                    long* alp = (long*)ap, blp = (long*)bp;
+                    for (; len >= 8; len -= 8)
+                    {
+                        if (*alp != *blp) return false;
+                        alp++;
+                        blp++;
+                    }
+                    byte* abp = (byte*)alp, bbp = (byte*)blp;
+                    for (; len > 0; len--)
+                    {
+                        if (*abp != *bbp) return false;
+                        abp++;
+                        bbp++;
+                    }
+                }
+            }
+            return true;
+        }
+
+        public static byte[] Skip(this byte[] a, int count)
```

Comment:
Why are you re-implementing the skip from `System.Linq`?
You've created an extension method for the byte[] which is an instance of Enumerable, basically the same as the default.
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skip?view=netcore-3.1#System_Linq_Enumerable_Skip__1_System_Collections_Generic_IEnumerable___0__System_Int32_

Reply:
Linq it's very slow for this class